homeassistant/components/eight_sleep/__init__.py

```diff
@@ -112,53 +107,70 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         # No users, cannot continue
         return False
 
-    hass.data[DOMAIN] = {
+    dev_reg = async_get(hass)
+    model = eight.device_data["modelString"]  # pylint: disable=unsubscriptable-object
+    hw_ver = eight.device_data["sensorInfo"][  # pylint: disable=unsubscriptable-object
+        "hwRevision"
+    ]
+    fw_ver = eight.device_data[  # pylint: disable=unsubscriptable-object
+        "firmwareVersion"
+    ]
```

Comment:
It would be nice to fix the upstream lib

Reply:
that's in progress: https://github.com/mezz64/pyEight/pull/27

Reply:
I also plan to move these properties out of a dict and into a dataclass or something else that can be more strongly typed